lesson13/script.js

```diff
@@ -0,0 +1,86 @@
+function showLoadingImg(){
+    const div = document.createElement('div');
+    const loadingImg = document.createElement('img');
+    div.id = "js-loading";
+    loadingImg.src = "img/loading-circle.gif";
+    loadingImg.alt = "ローディング画像";
+    document.body.insertAdjacentElement('afterbegin',div).appendChild(loadingImg);
+}
+
+function removeLoadingImg(){
+    document.getElementById('js-loading').remove();
+}
+
+async function fetchdata(){
+    const endpoint = "https://api.json-generator.com/templates/szdgGQcOLXuk/data?access_token=hu4bc7qh9znx2m8f53mn4mz2hryvdntkavwbw8j0";
+    const response = await fetch(endpoint);
+    if(!response.ok){
+    throw new Error(`${response.status}:${response.text}`);
+    }
+    const json = await response.json();
+    const data = await json.data;
+    return data;
+}
+
+function renderList(result) {
+    const fragment = document.createDocumentFragment();
+    const ul = document.getElementById('js-list');
+        for(const list of result){
+            const li = document.createElement('li');
+            li.className = "list-item";
+            const a = document.createElement('a');
+            const img = document.createElement('img');
+            a.textContent = list.text;
+            a.href = `/${list.a}`;
+            img.src = list.img;
+            img.alt = list.alt;
+            fragment.appendChild(li).appendChild(a).insertAdjacentElement('afterbegin',img);
+        }
+    ul.appendChild(fragment);
+}
+
+async function init(){
+    showLoadingImg();
+    try{
+        const result = await fetchdata();
+        if (!result) {
+            return;
+        }
+        renderList(result);
+    }catch(error){
+        document.getElementById('js-list').textContent = error;
+        console.error(error);
+    }finally{
+        removeLoadingImg();
+    }
+}
+
+const modal = document.getElementById('js-modal');
+const openButton = document.getElementById('js-open-btn');
+const closeButton = document.getElementById('js-close-btn');
+const requestButton = document.getElementById('js-request-btn');
+
+function openModalWindow(){   
+    openButton.addEventListener('click',function () {
+        modal.classList.add('is-show');
+        this.classList.add('is-deactive');
```

Comment:
Q
Why is `is-deactive`?
(Translated from Japanese `不活発/非活動/不活性`)

Reply:
I made it deactive because it means a button that cannot be used when the modal is open.
But it's a class name to hide the element, so it may not be appropriate.

Reply:
I changed the class name from is-deactive to is-hide.
https://github.com/nakagawamai/js-lesson/pull/15/commits/20ce95e6a66aaf37266dbdcc3a3898aac2d28f8f